tests/devices/eiger/test_eiger.py

```diff
@@ -0,0 +1,347 @@
+    mock_eiger = create_autospec(Eiger, instance=True)
+    mock_eiger.status = mock_status
+    mock_eiger.settings = mock_settings
+    return mock_eiger
+
+
+@pytest.fixture
+def raise_interrupt():
+    async def raise_interrupt():
+        return False
+
+    return Mock(raise_interrupt)
+
+
+@pytest.fixture
+def eiger_adapter(mock_eiger):
+    return EigerAdapter(mock_eiger, raise_interrupt, host="localhost", port=8081)
+
+
+def test_eiger_adapter_contructor():
+    EigerAdapter(mock_eiger, raise_interrupt, host="localhost", port=8081)
+
+
+# @pytest.fixture
+# async def mock_client() -> Mock:
+#     app = web.Application()
+#     client = await aiohttp_client(app)
+#     return client
+
+
+DETECTOR_API = "http://localhost:8081/detector/api/1.8"
+
+
+@pytest.fixture
+def mock_good_get_request():
+    mock_request = MagicMock(web.Request)
+    mock_request.match_info = {"parameter_name": "count_time"}
+
+    return mock_request
+
+
+@pytest.fixture
+def mock_bad_get_request():
+    mock_request = MagicMock(web.Request)
+    mock_request.match_info = {"parameter_name": "wrong_param"}
+
+    return mock_request
+
+
+@pytest.fixture
+def mock_good_put_request():
+    mock_request = MagicMock(web.Request)
+    mock_request.match_info = {"parameter_name": "count_time"}
+    mock_request.json.return_value = {"value": 0.5}
+
+    return mock_request
+
+
+@pytest.fixture
+def mock_bad_put_request():
+    mock_request = MagicMock(web.Request)
+    mock_request.match_info = {"parameter_name": "wrong_param"}
+    mock_request.json.return_value = {"value": 0.5}
+
+    return mock_request
+
+
+@pytest.fixture
+def mock_good_get_status_request():
+    mock_request = MagicMock(web.Request)
+    mock_request.match_info = {"status_param": "state"}
+
+    return mock_request
+
+
+@pytest.fixture
+def mock_bad_get_status_request():
+    mock_request = MagicMock(web.Request)
+    mock_request.match_info = {"status_param": "wrong_param"}
+
+    return mock_request
+
+
+@pytest.mark.asyncio
+async def test_eiger_get_config(eiger_adapter, mock_good_get_request):
+
+    resp = await eiger_adapter.get_config(mock_good_get_request)
+
+    assert isinstance(resp, web.Response)
+
+
+@pytest.mark.asyncio
+async def test_eiger_good_get_config(eiger_adapter, mock_good_get_request):
+
+    resp = await eiger_adapter.get_config(mock_good_get_request)
+
+    assert resp.text != "None"
+
+
+@pytest.mark.asyncio
+async def test_eiger_bad_get_config(eiger_adapter, mock_bad_get_request):
+
+    resp = await eiger_adapter.get_config(mock_bad_get_request)
+
+    assert resp.text == "None"
+
+
+@pytest.mark.asyncio
+async def test_eiger_good_put_config_wrong_device_state(
+    eiger_adapter, mock_good_put_request
+):
+
+    eiger_adapter._device.get_state.return_value = State.NA
+
+    resp = await eiger_adapter.put_config(mock_good_put_request)
+
+    assert isinstance(resp, web.Response)
+    assert resp.text == "Eiger not initialized or is currently running."
```

Comment:
Could subclass `web.Response` and fix the value of `text`; It'd be a bit more verbose but it would ensure stuff stays in sync

Reply:
In the main program

Reply:
> it would ensure stuff stays in sync

What do you mean exactly?